akka-tools-persistence/src/main/scala/no/nextgentel/oss/akkatools/persistence/EnhancedPersistentActor.scala

```diff
@@ -179,37 +261,47 @@ abstract class EnhancedPersistentActor[E:ClassTag, Ex <: Exception : ClassTag]
       }
     }
 
+    deletePotentialMessagesMarkedForDeletion()
+  }
+
+  private def deletePotentialMessagesMarkedForDeletion(): Unit = {
+    internalDurableState.deleteMessagesUpTo.foreach(deleteMark => {

```

Comment:
you only have to find the highest deletemark and then check if  you have to delete something once.

Reply:
I am looping an Optional, so it should happen only once I think.
I'll switch it to a pattern match though, more idiomatic Scala I think, and clearer.